x/valset-pref/client/cli/tx.go

```diff
@@ -26,19 +28,42 @@ func NewSetValSetCmd() *cobra.Command {
 	return osmocli.TxCliDesc{
 		Use:              "set-valset [delegator_addr] [validators] [weights]",
 		Short:            "Creates a new validator set for the delegator with valOperAddress and weight",
-		Example:          "osmosisd tx valset-pref set-valset osmo1... osmovaloper1abc...,osmovaloper1def...  0.56,0.44",
+		Example:          "osmosisd tx validatorsetpreference set-valset osmo1... osmovaloper1abc...,osmovaloper1def...  0.56,0.44",
 		NumArgs:          3,
 		ParseAndBuildMsg: NewMsgSetValidatorSetPreference,
 	}.BuildCommandCustomFn()
 }
 
+func NewDelValSetCmd() *cobra.Command {
+	return osmocli.TxCliDesc{
+		Use:              "delegate-valset [delegator_addr] [amount]",
+		Short:            "Delegate tokens to existing valset using delegatorAddress and tokenAmount.",
+		Example:          "osmosisd tx validatorsetpreference delegate-valset  osmo1... 100stake",
+		NumArgs:          2,
+		ParseAndBuildMsg: NewMsgDelegateToValidatorSet,
+	}.BuildCommandCustomFn()
+}
+
+func NewUnDelValSetCmd() *cobra.Command {
+	return osmocli.TxCliDesc{
+		Use:              "undelegate-valset [delegator_addr] [amount]",
+		Short:            "UnDelegate tokens from existing valset using delegatorAddress and tokenAmount.",
+		Example:          "osmosisd tx validatorsetpreference undelegate-valset  osmo1... 100stake",
+		NumArgs:          2,
+		ParseAndBuildMsg: NewMsgUndelegateFromValidatorSet,
+	}.BuildCommandCustomFn()
+}
+
```

Comment:
I dont think these need to be parsed with a custom build message (I could be wrong)

See this as an example:

https://github.com/osmosis-labs/osmosis/blob/05a9fe4262a5fcc302e98b0ddf87d9f5f538b05e/x/concentrated-liquidity/client/cli/tx.go#L25-L51

Osmocli should just be able to parse input to the message on its own

Reply:
That is so crazy!!! no need for parsing at all

Reply:
I know, big shoutout to @ValarDragon and chatGPT lol